### ISSUE TITLE
Icon Selector very laggy and cpu intensive

### ISSUE THREAD

**Issue by zhrexl** [Schneegans/Fly-Pie#316]
## Describe the Bug
When the icon selector is opened it gets very laggy, its really hard to select an icon. The CPU goes to 100% when scrolling the GtkIconView (Core i7 12700H).

1. Go to Menu Editor
2. Try to select a new icon for the menu

Here's a sysprof of the problem
![image](https://github.com/Schneegans/Fly-Pie/assets/51864789/1da820c3-59cd-4e3d-9bc8-20f6b6008117)

## Expected Behavior
Not be laggy at all just like others icon selectors.

If more information is needed I can try to gather it.

## System
 - Fedora 38
 - GNOME Shell version: 44

**Comment by Schneegans** [comment]
Yeah, I know that it can be laggy sometimes. The amount of "lagginess" depends on the icon theme you use. With the default Adwaita icons it's OK on my machine. Bigger icon sets can be more demanding.

However, I think there's no easy fix as the `GtkIconView` is simply too slow for this use-case. I could rewrite the dialog using a more modern widget such as the `GtkGridView` but this will make it difficult to remain backward-compatible. When GNOME 45 is released later this year, Fly-Pie will need some severe porting which will drop support for all older GNOME versions anyways. After that, it will be much easier to clean up some of the GTK3 legacy code in the UI.

**Comment by Schneegans** [comment]
I will not have the time to work on this as I am currently working on Fly-Pie's follow-up project [🌸 Kando](https://github.com/kando-menu/kando). If anybody wants to work on this, I would happily accept a pull request. In the mean time, you could try out Kando 😉
